src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -521,9 +542,83 @@ else if (job.getTree().hasLeaves() && job.getTree().isSlimeTree())
             }
             job.getTree().pollNextLeaf();
         }
+        else if (job.getTree().hasLeaves() && job.getTree().isNetherTree())
+        {
+            final BlockPos leaf = job.getTree().peekNextLeaf();
+            if (!mineBlock(leaf, workFrom))
+            {
+                return getState();
+            }
+            job.getTree().pollNextLeaf();
+        }
         return getState();
     }
 
+    /**
+     * Bonemeal fungi in netherTrees
+     */
+
+    private IAIState bonemealFungi()
+    {
+        for (final BlockPos pos : getNetherTrees())
+        {
+            final BlockState blockState = world.getBlockState(pos);
+            final Block block = blockState.getBlock();
+            if (block == Blocks.CRIMSON_FUNGUS || block == Blocks.WARPED_FUNGUS)
+            {
+                final int threshold = worker.getCitizenData().getCitizenSkillHandler().getLevel(getOwnBuilding().getPrimarySkill()) * FUNGI_MODIFIER;
+                final int rand = new Random().nextInt(100);
```

Comment:
avoid using new Random, instead use the colonies or world's random instance. Creating a new is kinda heavy